Global.ts

```diff
@@ -0,0 +1,13 @@
+import { IApp } from '@rocket.chat/apps-engine/definition/IApp';
+
+export class Global {
+    /*
+    * TODO: We are only using app for logger(Which shows up in Admin -> App Info -> Logs).
+    * It is being used in many core function of this app i.e, createMessage. Which makes these functions not usable where the app object is not available
+    * This Globle App object can be used in these functions for now. And also it's not a good idea to Global whole App object.
+    * So in future we will make only necessary object Global. i.e. Logger
+    * The idea is to remove dependecy of this App object from every core functions. and use Global logger instead.
+    * This is big refactor and will be done as separate task
+    */
+    public static app: IApp;
+}
```

Comment:
@AlexanderKanakis After this, our life will be peaceful 🕺. We can now have access to `App` in SchedulerProcessor which you were trying in https://github.com/WideChat/Rocket.Chat.Apps-engine/pull/47/commits/e26835c65c95f4724d34161fb146e66a6c586001

Can you please test and review whenever you get the chance?

Eventually, we will get rid of `App` from these functions, because as of now it's only being used for logging.

Reply:
This is great @Shailesh351! Before I review, could you tell me when onEnable is called?

Reply:
@AlexanderKanakis When App is installed and enabled https://developer.rocket.chat/apps-engine/getting-started/app-lifecycle#onenable

Reply:
@Shailesh351 can you make a story on our git board please about the refactor you have mentioned above?  Thanks.